H.Ws/H.W_4.py

```diff
@@ -0,0 +1,236 @@
+import numpy
+import pandas
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import confusion_matrix
+from sklearn.model_selection import train_test_split
+
+titanic_df = pandas.read_csv(
+    "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/titanic.csv"
+)  # .dropna().reset_index()
+
+print(titanic_df)
+print(titanic_df.dtypes)
+print(titanic_df.columns)
+predictors = ["pclass", "sex", "age", "sibsp", "embarked", "class"]
+response = ["alive"]
+
+x = titanic_df[predictors]
+y = titanic_df[response]
+resp_test = None
+pred_test = None
+
+# for response
+for val in response:
```

Comment:
You have the right idea checking if response variables are boolean or continuous. I think this won't catch the case where the response column is already in the form of ('y' or 'n'). It also does not deal with columns that are already in the form of (0,1).